### PR TITLE
VR: Enhance logging of whole operator

### PR DESCRIPTION
**Pull request by iamniting**
As part of this commit did below enhancements
- Add VRName in the logs at all places
- Add other object name at places where ever possible
- Add uniformity in the logs

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/volumereplication_controller.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			r.Log.Info("no VolumeReplication resource found")
```
Comment (ShyamsundarR) [inline controllers/volumereplication_controller.go:87]: (question) If you actually used the logger returned from line 77 it would print this information for all logs, without having to add this to each log line. Attempting to learn why that was not chosen.

Comment (iamniting) [reply controllers/volumereplication_controller.go:87]: Yeah, Nice question. line 77 did not caught my attention. Maybe we should use that. @umangachapagain @sp98 Can we use that or there was any specific reason to not use it?

**Comment by Madhu-1** [comment]
@umangachapagain PTAL

File: controllers/volumereplication_controller.go (hunk 1 above)
Comment (umangachapagain) [reply controllers/volumereplication_controller.go:87]: We should use that.

File: controllers/volumereplication_controller.go [hunk 2]
```diff
@@ -74,7 +74,9 @@ type VolumeReplicationReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.0/pkg/reconcile
 func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 
-	_ = r.Log.WithValues("controller_volumereplication", req.NamespacedName)
+	prevLogger := r.Log
```
Comment (Madhu-1) [inline controllers/volumereplication_controller.go:77]: why these changes are required?

File: controllers/volumereplication_controller.go (hunk 1 above)
Comment (iamniting) [reply controllers/volumereplication_controller.go:87]: Done

File: controllers/volumereplication_controller.go (hunk 2 above)
Comment (iamniting) [reply controllers/volumereplication_controller.go:77]: because of this reason https://github.com/openshift/ocs-operator/pull/990#discussion_r558330313

Moreover `VolumeReplicationReconciler` gets generated once when we start the operator and we are changing its value r.log field which will be retained even when it returns from reconciling. So it does not make sense to keep the requested value in r.log IMO.

File: controllers/secret.go [hunk 3]
```diff
@@ -32,6 +32,7 @@ func (r *VolumeReplicationReconciler) getSecret(name, namespace string) (map[str
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			r.Log.Error(err, "Secret not found", "Secret Name", name, "Secret Namespace", namespace)
```
Comment (sp98) [inline controllers/secret.go:34]:
```suggestion
			r.Log.Error(err, "secret not found", "Secret Name", name, "Secret Namespace", namespace)
```

lets keep every message in the same case (lower case) for consistency.

File: controllers/volumereplication_controller.go [hunk 4]
```diff
@@ -130,58 +132,58 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		_, pv, err := r.getPVCDataSource(nameSpacedName)
 		if err != nil {
 			_ = r.updateReplicationStatus(instance, replicationv1alpha1.ReplicationFailure, err.Error())
-			r.Log.Error(err, "failed to get dataSource for PVC", "dataSourceName", nameSpacedName.Name)
+			r.Log.Error(err, "failed to get PVC", "PVCName", instance.Spec.DataSource.Name)
```
Comment (sp98) [inline controllers/volumereplication_controller.go:135]: do you want to add instance name here as well?

File: controllers/volumereplication_controller.go [hunk 5]
```diff
@@ -130,58 +132,58 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
 		}
-		r.Log.Info("volumeReplication object is terminated, skipping reconciliation", "objectName", instance.Name)
+		r.Log.Info("volumeReplication object is terminated, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
 	instance.Status.LastStartTime = getCurrentTime()
 	if err = r.Client.Update(context.TODO(), instance); err != nil {
-		r.Log.Error(err, "Failed to update status", "Object", instance.Name)
+		r.Log.Error(err, "Failed to update status")
```
Comment (sp98) [inline controllers/volumereplication_controller.go:180]:
```suggestion
		r.Log.Error(err, "failed to update status")
```

File: controllers/volumereplication_controller.go [hunk 6]
```diff
@@ -130,58 +132,58 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
 		return reconcile.Result{}, err
 	}
 
 	// enable replication on every reconcile
 	if err = r.enableReplication(volumeHandle, parameters, secret); err != nil {
-		r.Log.Error(err, "Failed to enable replication", "Object", instance.Name)
+		r.Log.Error(err, "Failed to enable replication")
```
Comment (sp98) [inline controllers/volumereplication_controller.go:186]:
```suggestion
		r.Log.Error(err, "failed to enable replication")
```

**Review by Madhu-1 — APPROVED** [review]

File: controllers/volumereplication_controller.go [hunk 7]
```diff
@@ -238,7 +240,7 @@ func (r *VolumeReplicationReconciler) updateReplicationStatus(instance *replicat
 	instance.Status.State = state
 	instance.Status.Message = message
 	if err := r.Client.Status().Update(context.TODO(), instance); err != nil {
-		r.Log.Error(err, "Failed to update status", "Object", instance.Name)
+		r.Log.Error(err, "Failed to update status")
```
Comment (sp98) [inline controllers/volumereplication_controller.go:243]:
```suggestion
		r.Log.Error(err, "failed to update status")
```

**Review by sp98 — CHANGES_REQUESTED** [review]
nit

File: controllers/volumereplication_controller.go (hunk 4 above)
Comment (iamniting) [reply controllers/volumereplication_controller.go:135]: Nope, now we have a instance name by default

**Review by sp98 — CHANGES_REQUESTED** [review]
one last thing . Log in the `requeueForResync` if block   [here](https://github.com/csi-addons/volume-replication-operator/blob/a918564f6b70be59ba0600944ced11f81402d546/controllers/volumereplication_controller.go#L218) would be very useful.

**Comment by iamniting** [comment]
> one last thing . Log in the `requeueForResync` if block [here](https://github.com/csi-addons/volume-replication-operator/blob/a918564f6b70be59ba0600944ced11f81402d546/controllers/volumereplication_controller.go#L218) would be very useful.

done, updated status also

**Review by sp98 — APPROVED** [review]

File: controllers/volumereplication_controller.go [hunk 8]
```diff
@@ -200,19 +202,21 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		requeueForResync, replicationErr = r.resyncVolume(instance, volumeHandle, parameters, secret)
 
 	default:
-		replicationErr = fmt.Errorf("unsupported image state %q", instance.Spec.ReplicationState)
-		r.Log.Error(replicationErr, "unsupported image state", "Name", instance.Name, "ReplicationState", instance.Spec.ReplicationState)
+		replicationErr = fmt.Errorf("unsupported image state")
+		r.Log.Error(replicationErr, "given image state is not supported", "ReplicationState", instance.Spec.ReplicationState)
```
Comment (Madhu-1) [inline controllers/volumereplication_controller.go:206]: the image needs to be changed to volume.

Comment (iamniting) [reply controllers/volumereplication_controller.go]: Done
